config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -101,3 +107,193 @@ const (
+	// configure the persistent volume claim, including storage class, volume
+	// size, and name.
+	// If omitted, the Pod uses ephemeral storage and alert data will not persist
+	// across restarts.
+	// // This field is optional.
+	// +optional
+	VolumeClaimTemplate v1.PersistentVolumeClaim `json:"volumeClaimTemplate,omitempty"`
+}
+
+// SecretName is a type that represents the name of a Secret in the same namespace.
+// +kubebuilder:validation:MaxLength=253
+type SecretName string
+
+// AlertManagerDeployMode defines the deployment state of the platform Alertmanager instance.
+//
+// Possible values:
+// - "Deployed": The Alertmanager instance will be deployed and managed by the operator.
+// - "NotDeployed": The operator will not deploy an Alertmanager instance.
+type AlertManagerDeployMode string
+
+const (
+	// AlertManagerModeEnabled means the Alertmanager instance will be deployed and managed by the operator.
+	AlertManagerManagerDeployed AlertManagerDeployMode = "Deployed"
+
+	// AlertManagerModeDisabled means the operator will not deploy the Alertmanager instance.
+	AlertManagerModeNotDeployed AlertManagerDeployMode = "NonDeployed"
+)
+
+// UserAlertManagerMode defines mode for user-defines namespaced
+//
+// Possible values:
+// - "Selectable": User-defined namespaces can be selected for AlertmanagerConfig lookups.
+// - "None": User-defined namespaces cannot be selected for AlertmanagerConfig lookups.
+type UserAlertManagerMode string
+
+const (
+	// UserAlertmanagerEnabled enables user-defined namespaces to be selected for `AlertmanagerConfig` lookups. This setting only
+	// applies if the user workload monitoring instance of Alertmanager is not enabled.
+	UserAlertManagerSelectable UserAlertManagerMode = "Selectable"
+	// UserAlertManagerDisabled disables user-defined namespaces to be selected for `AlertmanagerConfig` lookups. This setting only
+	// applies if the user workload monitoring instance of Alertmanager is not enabled.
+	UserAlertManagerNone UserAlertManagerMode = "None"
+)
+
+// +kubebuilder:validation:Enum=Error;Warn;Info;Debug
+type LogLevel string
+
+const (
+	LogLevelError LogLevel = "Error"
+	LogLevelWarn  LogLevel = "Warn"
+	LogLevelInfo  LogLevel = "Info"
+	LogLevelDebug LogLevel = "Debug"
+)
+
+// ResourceSpec defines the requested and limited value of a resource.
+type ResourceSpec struct {
+	// request is the minimum amount of the resource required.
+	// +kubebuilder:validation:MaxLength=8
+	// +optional
+	Request string `json:"request,omitempty"`
+	// limit is the maximum amount of the resource allowed.
+	// +kubebuilder:validation:MaxLength=8
+	// +optional
+	Limit string `json:"limit,omitempty"`
+}
+
+// AlertmanagerContainerResources defines simplified resource requirements for a container.
+type AlertmanagerContainerResources struct {
+	// cpu defines the CPU resource limits and requests.
+	// Format: "<request>,<limit>" (e.g. "100m,500m")
+	// +optional
```

Comment:
Mention explicitly that the field is optional in the GoDoc.

Reply:
done